### PR TITLE
Feature/issue-5-twistd-plugin

### PR DESCRIPTION
**Pull request by CarlaLlama**
FINISHED?

Twisted plugin working, command:

`twistd -n vumi_http_proxy`

`twistd -n openross` to see list of all plugins and commands

@hodgestar

### CODE REVIEW

File: twisted/plugins/dropin.cache [hunk 1]
```diff
@@ -0,0 +1,48 @@
+(dp1
```
Comment (hodgestar) [inline twisted/plugins/dropin.cache]: Oops. This should be `git rm`-ed now that it's been added to `.gitignore`.

File: vumi_http_proxy/http_proxy.py [hunk 2]
```diff
@@ -38,7 +38,6 @@ class Proxy(proxy.Proxy):
 
     def __init__(self, blacklist):
         proxy.Proxy.__init__(self)
-        self.blacklist = blacklist
```
Comment (hodgestar) [inline vumi_http_proxy/http_proxy.py:41]: Was this change intentional?

File: twisted/plugins/dropin.cache (hunk 1 above)
Comment (CarlaLlama) [inline twisted/plugins/dropin.cache]: Thanks!

File: vumi_http_proxy/servicemaker.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def makeService(self, options):
+        """ Call Initialize to start endpoint server """
+        application = Application("basic AMP server")
+
+        factory = ProxyFactory(["asdf.com"])
+        endpoint = serverFromString(
+            reactor, "tcp:%d:interface=%s" % (
+                options["port"], str(["self.ip"])))
+        reactor.run()
```
Comment (hodgestar) [inline vumi_http_proxy/servicemaker.py]: I think we can delete `reactor.run()` here -- twistd's infrastructure will do the equivalent for us and start the service we return.

File: vumi_http_proxy/http_proxy.py (hunk 2 above)
Comment (CarlaLlama) [inline vumi_http_proxy/http_proxy.py:41]: I removed it in a desperate attempt when I was getting an "cannot connect, already running" error, and forgot to put back

File: vumi_http_proxy/servicemaker.py (hunk 3 above)
Comment (CarlaLlama) [inline vumi_http_proxy/servicemaker.py]: Oh okay that's cool

Comment (CarlaLlama) [inline vumi_http_proxy/servicemaker.py]: There's supposed to be an application/parent in the brackets of .setServiceParent() - but I'm not sure what

File: twisted/plugins/proxy_worker_starter.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from vumi_http_proxy.servicemaker import ProxyWorkerServiceMaker
+
+proxy_worker = ProxyWorkerServiceMaker()
```
Comment (hodgestar) [inline twisted/plugins/proxy_worker_starter.py]: Should we perhaps name this after the proxy package? So `vumi_http_proxy`?

**Comment by hodgestar** [comment]
And we should write a test for the servicemaker.

File: vumi_http_proxy/test/test_servicemaker.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        options.parseOptions(["--interface", "127.0.0.1"])
+        self.assertEqual(options["port"], "8000")
+        self.assertEqual(str(options["interface"]), "127.0.0.1")
+
+
+class TestProxyWorkerServiceMaker(unittest.TestCase):
+    def test_makeService(self):
+        self.assertEqual(
+          str(ProxyWorkerServiceMaker.makeService),
+          '<unbound method ProxyWorkerServiceMaker.makeService>')
```
Comment (hodgestar) [inline vumi_http_proxy/test/test_servicemaker.py]: We should actually call `makeService` (with some options) and then assert things about the service we get back, e.g.:

``` python
    options = Options()
    options.parseOptions([])
    servicemaker = ProxyWorkerServiceMaker()
    service = servicemaker.makeService(options)
    self.assertEqual(service.factory, vumi_http_proxy.http_proxy.ProxyFactory)
    self.assertEqual(service.endpoint._interface, '0.0.0.0')
    self.assertEqual(service.endpoint._port, 8080)
```

**Comment by hodgestar** [comment]
One comment about the service maker test, but otherwise looks shiny!

**Comment by CarlaLlama** [comment]
Is this better?

**Comment by hodgestar** [comment]
:+1:
